app/js/questionNav.js

```diff
@@ -66,25 +67,19 @@ function changeQuestion(destinationPage) {
 function fillNav() {
     let nav = document.querySelector("#question-nav")
     let questions = document.querySelectorAll('.question')
-    let counter = 1
     questions.forEach(function (question) {
-        nav.innerHTML += '<span class="nav-item unanswered-nav-box"><p>' + question.dataset['id'] + '</p></span>'
-        counter++
-    })
-}
-
-/**
- * Makes nav-items clickable to jump to specified question
- */
-function addNavLinks() {
-    document.querySelectorAll('.nav-item').forEach(function(button) {
-        let qID = button.querySelector('p').textContent
-        button.addEventListener('click', function () {
-            changeQuestion(qID)
-            current = qID
+        let navItem = document.createElement('div')
+        navItem.classList.add('nav-item')
+        navItem.classList.add('unanswered-nav-box')
+        navItem.textContent = question.dataset['id']
+        navItem.addEventListener('click', function () {
+            changeQuestion(question.dataset['id'])
+            current = question.dataset['id']
```

Comment:
why not do this on the line above then pass it in, instead of accessing the dataset variable twice?

Reply:
changed in 5bfc0ca